src/main/java/uk/gov/justice/digital/hmpps/keyworker/services/PrisonStatsService.kt

```diff
@@ -24,10 +24,12 @@ class PrisonStatsService(
     prisonCode: String,
     from: LocalDate,
     to: LocalDate,
+    comparisonFrom: LocalDate?,
+    comparisonTo: LocalDate?,
   ): PrisonStats {
     val dateRange = DAYS.between(from, to.plusDays(1))
-    val previousTo = from.minusDays(1)
-    val previousFrom = from.minusDays(dateRange)
+    val previousTo = comparisonTo ?: from.minusDays(1)
```

Comment:
If either comparisonTo or comparisonFrom are null, both previousTo and previousFrom should be calculated. This will use combinations of supplied comparison dates and initial dates

Reply:
Yeah that's true but the staff path does account for it. Also one uses ReportingPeriod and the other doesn't. Neither is an issue so it's up to you if you want to change it or do any form of a refactor to make these ranges uses common code